fastify.js

```diff
@@ -461,6 +463,7 @@ function override (old, fn, opts) {
   instance[kHooks] = buildHooks(instance[kHooks])
   instance[kRoutePrefix] = buildRoutePrefix(instance[kRoutePrefix], opts.prefix)
   instance[kLogLevel] = opts.logLevel || instance[kLogLevel]
+  instance[kLogSerializers] = Object.assign(Object.create(instance[kLogSerializers]), opts.logSerializers)
```

Comment:
In the usual case, `Object.assign(Object.create(null), null)` will return `{ }`.
So I would write:

```suggestion
if(instance[kLogSerializers] || opts.logSerializers) {
  instance[kLogSerializers] = Object.assign(Object.create(instance[kLogSerializers]), opts.logSerializers)
}
```

Reply:
I understand your point, but checking two address memory `instance[kLogSerializers]` and `opts.logSerializers` is not more overhead than `{}`?

Reply:
In this case, I think we are comparing RAM access vs RAM allocation.
Moreover, this code is run only at startup, so the IF will be executed one time while the object will remain in memory during all the server lifetime that will impact the Node.js GC.
So: if I have 100 routes, I will have 1 `{ }` in memory for each context plus, at every request will allocate another `{ }`.. and so on.

I don't have data about it to understand better the performance impact... we could produce a case study, it would be interesting for me to understand deeper node.js 👍